util/MsSql/backup-db.sh

```diff
@@ -1,7 +1,11 @@
 #!/bin/sh
 
-# Delete backup files older than 30 days
-find /etc/bitwarden/mssql/backups/ -type f -name '*.BAK' -mindepth 1 -mtime +30 -delete
+# Backup timestamp
+export now=${1:-$(date +%Y%m%d_%H%M%S)}
 
 # Do a new backup
 /opt/mssql-tools/bin/sqlcmd -S localhost -U sa -P ${SA_PASSWORD} -i /backup-db.sql
+
+# Delete backup files older than 30 days
+grep -B1 "BACKUP DATABASE successfully" /var/opt/mssql/log/errorlog | grep -q _$now.BAK &&
```

Comment:
What does this `-B1` option do?

Reply:
`-B1` means one line back.
When a backup succeeds, log looks like this :
```
2018-05-29 06:03:58.53 Backup      Database backed up. Database: vault, creation date(time): 2018/03/05(08:26:55), pages dumped: 22615, first LSN: 70:17104:37, last LSN: 70:17136:1, number of dump devices: 1, device information: (FILE=1, TYPE=DISK: {'/etc/bitwarden/mssql/backups/vault_FULL_20180529_080349.BAK'}). This is an informational message only. No user action is required.
2018-05-29 06:03:58.57 Backup      BACKUP DATABASE successfully processed 22602 pages in 8.556 seconds (20.637 MB/sec).
```

When then look for `BACKUP DATABASE successfully` and the line before (`-B1`).
Finally, in these 2 lines, we look for the backup name (second `grep`).